app/views/studies/_shared_sync_functions.js.erb

```diff
@@ -1,9 +1,81 @@
 // shared JS functions for sync forms
 
-$('#study-file-<%= study_file.id %>').on('change', '.file-type', function() {
+// Used for sync CSFV
+if (typeof window.SCP.readOnlyToken === 'undefined') {
```

Comment:
Blocker: we will have an issue with this feature not working in private studies where the user is not registered for Terra.  Since this is a "streaming" scenario, we can't simply use the user's regular OAuth token from `window.SCP.userAccessToken`, because if the user isn't registered for Terra, that token will not work.  However, `get_read_access_token` will return `nil` if the study is private and the user is not registered for Terra, and all fetch calls will fail with `401` and display the "Your session may have timed out" modal.

I'm not completely sure what the right thing to do here is, but my feeling is that we should use `window.SCP.userAccessToken` since this is already registered in the DOM, and we don't need to interpolate it like we do with `window.SCP.readOnlyToken` in this case.  If the user is not registered for Terra, we should display a notice that client-side validation will not work due to the incomplete Terra profile, and direct them to `app.terra.bio` to complete their registration.  We would then disable CSFV on the sync page.

I can also see just checking for `window.SCP.readOnlyToken` to be present, and the doing the same as above, but I think using the regular token is slightly cleaner, and we'd just do a check on `current_user.registered_for_firecloud`.

Reply:
This is a great technical find, but I don't think it should block merge.  Your data up top elevates the discussion:

> this feature will break on private studies where the user is not registered for Terra. This comprises ~11% of all studies right now (124 of 1152)

A more precise way to assess impact is "_How many users_ not registered for Terra synced for private studies?"  Over the last year, [only 1 user not registered for Terra has visited the Sync page](https://mixpanel.com/s/2ghU35), and it's unclear if that was even a private study.

That's < 0.01% of our users.  The very low figure makes sense, since sync is premised on transferring data from a Terra workspace, and having a user who isn't registered for Terra do that seems like a very rare use case.

Given the vanishingly small impact, I think this refinement makes sense to consider as a separate task.